### PR TITLE
Disable registering 3D physics types when 3D is disabled

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR organizes the 2D and 3D physics code in `register_server_types.cpp` and allows the 3D physics type registration code to be disabled by wrapping it in `#ifndef _3D_DISABLED`.

In the current master, a macOS arm64 template_release build is 66.5 MB (66503407 bytes). Using `disable_3d=yes` brings this down to 60.2 MB (60224255 bytes), a reduction of 9.4%.

With this PR, the `disable_3d=yes` build is 58.8 MB (58848751 bytes), a reduction of 2.3% compared to master with this option, and brings the total savings from 9.4% to 11.5% compared to 3D enabled, a reduction of 2.1%.

### CODE REVIEW

**Comment by MewPurPur** [comment]
_poke poke?_

File: servers/register_server_types.cpp [hunk 1]
```diff
@@ -92,8 +83,22 @@
 #include "xr/xr_positional_tracker.h"
 #include "xr_server.h"
 
+#include "navigation_server_2d.h"
```
Comment (akien-mga) [inline servers/register_server_types.cpp:86]:
```suggestion
// 2D physics.
#include "navigation_server_2d.h"
```

File: servers/register_server_types.cpp [hunk 2]
```diff
@@ -92,8 +83,22 @@
...
+#include "physics_2d/godot_physics_server_2d.h"
+#include "physics_server_2d.h"
+#include "physics_server_2d_wrap_mt.h"
+#include "servers/extensions/physics_server_2d_extension.h"
+
+#ifndef _3D_DISABLED
```
Comment (akien-mga) [inline servers/register_server_types.cpp:92]: I think we should make explicit with a comment why those two blocks are separate from the rest.
```suggestion
// 3D physics.
#ifndef _3D_DISABLED
```

**Review by akien-mga — APPROVED** [review]

File: servers/register_server_types.cpp [hunk 3]
```diff
@@ -60,14 +60,7 @@
 #include "movie_writer/movie_writer.h"
 #include "movie_writer/movie_writer_mjpeg.h"
 #include "movie_writer/movie_writer_pngwav.h"
-#include "navigation_server_2d.h"
 #include "navigation_server_3d.h"
```
Comment (akien-mga) [inline servers/register_server_types.cpp:63]: I would also move this in the "3D physics + navigation" block, and just keep it outside the `#ifndef _3D_DISABLED` (with a comment justifying it's needed for 2D).

**Review by akien-mga — APPROVED** [review]

**Comment by AThousandShips** [comment]
This caused a regression, looking into a fix right now:
* https://github.com/godotengine/godot/issues/88666

**Comment by aaronfranke** [comment]
Apologies for the regression, I try to test these PRs thoroughly when I make them but it can be hard to keep up with testing after rebasing. I'm glad it was resolved quickly though.

**Pull request merged**